Problem Sets/extras.rkt

```diff
@@ -90,39 +90,66 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-;; check-location : String String -> Void
-;; GIVEN: a 2 digit problem set number NN and the name of the file
-;; being qualified.
-;; EFFECT: throws an error if this file is not a directory of the form
-;; pdp-*./setNN 
-(define (check-location NN correct-file-name)
-  (define path-elements (explode-path (current-directory)))
+;; (check-location NN correct-file-name) : module-level form
+(define-syntax (check-location stx)
+  (syntax-case stx ()
+    [(_ NN correct-file-name)
+     (case (syntax-local-context)
+       [(module)
+        #`(check-location-actual (#%expression NN)
+                                 (#%expression correct-file-name)
+                                 (quote #,(syntax-source stx)))
+        ]
+       [(top-level)
+        (raise-syntax-error #f
+          "`check-location' may only be used inside a module"
+          stx)]
+       [else
+        (raise-syntax-error #f
+          "found a use of `check-location' that is not at the top level"
```

Comment:
While it is true, that message would not offer any guidance on how to fix the problem.